### PR TITLE
CRM-16189, added function to validate toggling of deferred revenue ch…

### PR DESCRIPTION
**Pull request by pradpnayak**
…eck box

---
- CRM-16189: Improve support for Accrual Method bookkeeping
  https://issues.civicrm.org/jira/browse/CRM-16189

### CODE REVIEW

File: CRM/Financial/BAO/FinancialAccount.php [hunk 1]
```diff
@@ -246,4 +246,65 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
...
 
+  /**
+   * Validate if Deferred Account is set for Financial Type
+   * when Deferred Revenue is enabled
+   *
+   * @return string
+   *
+   */
+  public static function validateTogglingDeferredRevenue() {
+    $deferredFinancialType = self::getDeferredFinancialType();
```
Comment (eileenmcnaughton) [inline CRM/Financial/BAO/FinancialAccount.php]: my understanding is that you need to do the translate at this point for the translators to be able to define the strings

Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: Added the changes

**Comment by eileenmcnaughton** [comment]
Depends on https://github.com/civicrm/civicrm-core/pull/8568

File: CRM/Financial/BAO/FinancialAccount.php [hunk 2]
```diff
@@ -246,4 +246,65 @@ public static function getFinancialAccountForFinancialTypeByRelationship($financ
...
+    $message = ' Please revise the error message to the following:
```
Comment (JoeMurray) [inline CRM/Financial/BAO/FinancialAccount.php]: Please remove 'Please revise the error message to the following:' from the error message itself!

Comment (pradpnayak) [inline CRM/Financial/BAO/FinancialAccount.php]: Its removed in later commit.

**Comment by pradpnayak** [comment]
jenkins, test this please

**Comment by pradpnayak** [comment]
Rebased PR

**Comment by colemanw** [comment]
@pradpnayak this test failure looks relevant.

**Comment by pradpnayak** [comment]
@colemanw I have fixed the CRM test

**Pull request merged**